build-system/dep-check-config.js

```diff
@@ -45,6 +45,13 @@ exports.rules = [
       'extensions/amp-bind/0.1/bind-impl.js->src/sanitizer.js',
     ],
   },
+  {
+    filesMatching: '**/*.js',
+    mustNotDependOn: 'src/module.js',
```

Comment:
Do you think this is helpful, or unnecessary? I thought it might make sense to explicitly opt-in to this pattern.